node/src/test/kotlin/net/corda/node/internal/artemis/UserValidationPluginTest.kt

```diff
@@ -62,28 +68,13 @@ class UserValidationPluginTest {
 
     @Test(timeout = 300_000)
     fun `reject message with exception`() {
-        coreMessage.putStringProperty("_AMQ_VALIDATED_USER", BOB_NAME.toString())
-        val messageWithException = object : AMQPMessage(0, amqpMessage.buffer.array(), null) {
-            override fun getStringProperty(key: SimpleString?): String {
-                throw IllegalStateException("My exception")
-            }
-        }
+        val messageWithException = mock<AMQPMessage>()
+        whenever(messageWithException.getStringProperty(any<SimpleString>())).thenThrow(IllegalStateException("My exception"))
         // Artemis swallows all exceptions except ActiveMQException, so making sure that proper exception is thrown
         Assertions.assertThatExceptionOfType(ActiveMQSecurityException::class.java).isThrownBy {
             plugin.beforeSend(session, rigorousMock(), messageWithException, direct = false, noAutoCreateQueue = false)
         }.withMessageContaining("Message validation failed")
     }
 
-    @Test(timeout = 300_000)
-    fun `reject message with security exception`() {
-        coreMessage.putStringProperty("_AMQ_VALIDATED_USER", BOB_NAME.toString())
-        val messageWithException = object : AMQPMessage(0, amqpMessage.buffer.array(), null) {
-            override fun getStringProperty(key: SimpleString?): String {
-                throw ActiveMQSecurityException("My security exception")
-            }
-        }
-        Assertions.assertThatExceptionOfType(ActiveMQSecurityException::class.java).isThrownBy {
-            plugin.beforeSend(session, rigorousMock(), messageWithException, direct = false, noAutoCreateQueue = false)
-        }.withMessageContaining("My security exception")
-    }
+    private fun toAMQPMessage(message: ClientMessageImpl): AMQPMessage = AMQPConverter.getInstance().fromCore(message, NullStorageManager())
```

Comment:
Minor: If we are refactoring this code to a separate method from property getter, I would make it an extension method, as in:
```suggestion
    private fun ClientMessageImpl.toAMQPMessage(): AMQPMessage = AMQPConverter.getInstance().fromCore(this, NullStorageManager())
```

Reply:
sure will change